### ISSUE TITLE
More options for Config 

### ISSUE THREAD

**Issue by PerthSteve** [netson/ubuntu-unattended#40]
Hi, I’m quite new to this so sorry if it’s already been asked. 
I’ve got a headless server i with to install Ubuntu on but it needs to be completely automated from boot to login. This means I need to be able to give it a static IP and include ssh packages to let me login to it when it’s finished. From the readme it seems like I need a screen and kb attached to be able to run start.sh, which defeats the purpose of creating the automated install in the first place. Is this the case, or will this script do what is required for me?
Thanks!

**Comment by netson** [comment]
The script was initially created for my personal needs, which was to install ubuntu on a hypervisor, using a console. Without at least a console, you won't be able to select the "unattended install" option from the main menu. Also, configuring networking is currently not supported, other than DHCP. The point is not so much to configure your server (there are much better tools for that: ansible, puppet, chef, etc.) but simply to skip all the annoying install questions you get during the install and to load a script at the end of the process which can kick-off any post install tasks you may want to perform, such as installing/enabling OpenSSH. I intend to limit this script to a 100% original Ubuntu ISO and by enabling OpenSSH out of the box, or any other software, it wouldn't be 100% original ubuntu anymore.

I don't know anything about your infrastructure of course, but I use a combination of OpenStack and Ansible to do exactly what you're looking to do. Those tools allow me to spin up a virtual server and configure it from start to finish with a single command. Amazon EC2 and other suppliers support similar solutions.

Long story short, you will need at least a console to use this image, which I assume most if not all headless servers have (either through IPMI or similar, or through the hypervisor). If you're attempting to use this image on bare metal, you're likely not going to succeed. If you tell me a little more about what it is you're trying to achieve, I may be able to give you some pointers.

**Comment by PerthSteve** [comment]
Thanks for the details, I understand now. 
I’ve got an old dell t300 as a running system, but I’m putting an ssd in as the system disk so it’s an opportunity for me to upgrade and clean up by starting with a fresh install. I was doing a lot of googling for headless automated install and this script came up. I thought it was a full install but it looks like I’ll have to do a bit more reading. I just don’t have a screen, but if it’s too hard to automate fully I might have to lug this beast into my lounge and use the tv :-)
I’ve read there are ways to do it, possibly using an FTP server.

**Comment by Dhull442** [comment]
By any chance if you're asking as to how to remove that initial boot menu from the Installation, check [this](https://github.com/Dhull442/Unattended-ubuntu16.04-install#automating-the-boot-menu) out.
